geoapps_utils/driver/driver.py

```diff
@@ -50,6 +78,9 @@ def params(self):
 
     @params.setter
     def params(self, val: BaseParams | BaseData):
+        if isinstance(val, (str, Path, InputFile)):
```

Comment:
I am not a fan of this, we are mixing what params can be. Feels like you want to have an `input_file` setter instead, and do the setting there instead.